src/lib/aqua.ts

```diff
@@ -202,7 +200,46 @@ export const compileToFiles = async ({
       },
     ),
   );
-};
+}
+
+export async function compileFunctionCall(args: CompileFuncCallFromPathArgs) {
+  const { compileAquaCallFromPath } = await import("@fluencelabs/aqua-api");
+  return compileDirOrFile(compileAquaCallFromPath, args);
+}
+
+async function compileDirOrFile<
+  T extends { filePath: string },
+  U extends { errors: string[] },
+>(fn: (args: T) => Promise<U>, compileArgs: T) {
+  const isDir = (await stat(compileArgs.filePath)).isDirectory();
+
+  const compilationResultsWithFilePaths = isDir
+    ? await Promise.all(
+        (await getAquaFilesRecursively(compileArgs.filePath)).map(
+          async (aquaFilePath) => {
+            return {
+              compilationResult: await fn({
+                ...compileArgs,
+                filePath: aquaFilePath,
+              }),
+              aquaFilePath,
+            };
+          },
+        ),
+      )
+    : [
+        {
+          compilationResult: await fn(compileArgs),
+          aquaFilePath: compileArgs.filePath,
+        },
+      ];
```

Comment:
I don't like the design choice for `compileDirOrFile` function.
I suggest:
1. Put both imports from `aqua-api` inside `compileDirOrFile`
2. Remove `compileFunctionCall` altogether
3. Make 2 function override signature for `compileDirOrFile` - when it's called with aquaCall and just aqua. 
4. After these modifications function could be removed

Another alternative option i see fit is to make 2 separate functions for compiling aqua and aqua call and put shared logic in some helper function.

The reason for this is that passing concrete function into generic function is unnesessary abstraction which obscure the main logic and also makes code harder to read bcs you need to figure out what's called and when.

Implementation notes
```
type compileDirOrFileReturnType = Promise<
  [
    { compilationResult: CompilationResult; aquaFilePath: string }[],
    { compilationResult: CompilationResult; aquaFilePath: string }[],
  ]
>;
type compileDirOrFileReturnType2 = Promise<
  [
    { compilationResult: CallCompilationResult; aquaFilePath: string }[],
    { compilationResult: CallCompilationResult; aquaFilePath: string }[],
  ]
>;

async function compileDirOrFile(
  compileArgs: CompileFuncCallFromPathArgs,
): compileDirOrFileReturnType2;
async function compileDirOrFile(
  compileArgs: CompileFromPathArgs,
): compileDirOrFileReturnType;
async function compileDirOrFile(
  compileArgs: CompileFromPathArgs | CompileFuncCallFromPathArgs,
) {
  const { compileFromPath, compileAquaCallFromPath } = await import(
    "@fluencelabs/aqua-api"
  );

  const fn =
    "funcCall" in compileArgs
      ? (filePath?: string) => {
          return compileAquaCallFromPath({
            ...compileArgs,
            ...(filePath !== undefined ? { filePath } : {}),
          });
        }
      : (filePath?: string) => {
          return compileFromPath({
            ...compileArgs,
            ...(filePath !== undefined ? { filePath } : {}),
          });
        };

  const isDir = (await stat(compileArgs.filePath)).isDirectory();

  const compilationResultsWithFilePaths = isDir
    ? await Promise.all(
      (await getAquaFilesRecursively(compileArgs.filePath)).map(
        async (aquaFilePath) => {
          return {
            compilationResult: await fn(aquaFilePath),
            aquaFilePath,
          };
        },
      ),
    )
    : [
      {
        compilationResult: await fn(),
        aquaFilePath: compileArgs.filePath,
      },
    ];

  return splitErrorsAndResults(compilationResultsWithFilePaths, (result) => {
    return result.compilationResult.errors.length === 0
      ? { result }
      : { error: result };
  });
}
```

Reply:
We decided to leave this discussion to after MVM. Added TODO